tests/test_pipeline.py

```diff
@@ -32,6 +49,159 @@
+        """
+        self.element = element
+
+    def fit(self, X: Any, y: Any) -> Self:  # pylint: disable=invalid-name
+        """Fit the data.
+
+        Parameters
+        ----------
+        X : Any
+            The input data.
+        y : Any
+            The target data.
+
+        Returns
+        -------
+        Any
+            The fitted data.
+        """
+        self.element.fit(X, y)
+        return self
+
+    def transform(self, X: Any) -> Any:  # pylint: disable=invalid-name
+        """Transform the data.
+
+        Transform is called during prediction, which is not cached.
+        Since the transformer is not cached, the counter is not increased.
+
+        Parameters
+        ----------
+        X : Any
+            The input data.
+
+        Returns
+        -------
+        Any
+            The transformed data.
+        """
+        return self.element.transform(X)
+
+    def fit_transform(self, X: Any, y: Any) -> Any:  # pylint: disable=invalid-name
+        """Fit and transform the data.
+
+        Parameters
+        ----------
+        X : Any
+            The input data.
+        y : Any
+            The target data.
+
+        Returns
+        -------
+        Any
+            The transformed data.
+        """
+        global N_TRANSFORMATIONS  # pylint: disable=global-statement
+        N_TRANSFORMATIONS += 1
+        return self.element.fit_transform(X, y)
+
+    def get_params(self, deep: bool = True) -> dict[str, Any]:
+        """Get the parameters of the transformer.
+
+        Parameters
+        ----------
+        deep : bool
+            If True, the parameters of the transformer are also returned.
+
+        Returns
+        -------
+        dict[str, Any]
+            The parameters of the transformer.
+        """
+        params = {
+            "element": self.element,
+        }
+        if deep:
+            params.update(self.element.get_params(deep))
+        return params
+
+    def set_params(self, **params: Any) -> Self:
+        """Set the parameters of the transformer.
+
+        Parameters
+        ----------
+        **params
+            The parameters to set.
+
+        Returns
+        -------
+        Self
+            The transformer with the set parameters
+        """
+        element = params.pop("element", None)
+        if element is not None:
+            self.element = element
+        self.element.set_params(**params)
+        return self
+
+
+def _get_rf_regressor() -> Pipeline:
+    """Get a morgan + physchem + random forest pipeline.
+
+    To make the run extra slow, a TauTomerCanonicalizer is added.
+
+    Returns
+    -------
+    Pipeline
+        A pipeline with a morgan fingerprint, physchem descriptors, and a random forest
+    """
+    smi2mol = SmilesToMol()
+
+    global N_TRANSFORMATIONS  # pylint: disable=global-statement
+    N_TRANSFORMATIONS = 0
+
+    mol2concat = CountingTransformerWrapper(
+        MolToConcatenatedVector(
```

Comment:
We are in the unitstest here. It would be better to keep the number of features and feature computations to a minimum.

Reply:
Done